wayland-display-core/src/comp/input.rs

```diff
@@ -23,14 +24,14 @@ use std::{
     path::Path,
     time::Instant,
 };
+use smithay::input::keyboard::Keysym;
 
 pub struct NixInterface;
 
 impl LibinputInterface for NixInterface {
     fn open_restricted(&mut self, path: &Path, flags: i32) -> Result<OwnedFd, i32> {
-        fcntl::open(path, OFlag::from_bits_truncate(flags), stat::Mode::empty())
-            .map(|fd| unsafe { OwnedFd::from_raw_fd(fd) })
-            .map_err(|err| err as i32)
+        open(path, OFlags::from_bits_truncate(flags as u32), Mode::empty())
+            .map_err(|err| todo!("Convert Errno to i32?"))
```

Comment:
This is unfortunately just me being not much confident in Rust, I'm sure it's easily done..

Reply:
```suggestion
            .map_err(|err| err.raw_os_error())
```